pkg/bytecode/value.go

```diff
@@ -149,6 +166,18 @@ func (a arrayVal) Set(idx, val value) error {
 	return nil
 }
 
+func (a arrayVal) Slice(start, end value) (value, error) {
+	startIdx, endIdx, err := normalizeSliceIndices(start, end, len(a.Elements))
+	if err != nil {
+		return nil, err
+	}
+	elements := make([]value, endIdx-startIdx)
+	for i := startIdx; i < endIdx; i++ {
+		elements[i-startIdx] = a.Elements[i]
```

Comment:
From my understanding we don't need to use [copyOrRef](https://github.com/evylang/evy/blob/main/pkg/evaluator/value.go#L242) here because the composite types will pass by ref and the primitives in this file are aliased onto go primitives, so will be copied by value automatically.

Reply:
That sounds about right. The evaluator could probably be refactored to do values the same way as you're doing here and then copyOrRef would not be needed there.